src/test/java/com/microsoft/jenkins/azuread/integrations/casc/ConfigAsCodeCertificateTest.java

```diff
@@ -0,0 +1,126 @@
+import com.microsoft.jenkins.azuread.PermissionEntry;
+import hudson.model.Computer;
+import hudson.model.Item;
+import hudson.model.Node;
+import hudson.security.AuthorizationStrategy;
+import hudson.security.SecurityRealm;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.ConfiguratorRegistry;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.model.CNode;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.matrixauth.inheritance.NonInheritingStrategy;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.LoggerRule;
+
+import java.util.logging.Level;
+
+import static io.jenkins.plugins.casc.misc.Util.getJenkinsRoot;
+import static io.jenkins.plugins.casc.misc.Util.toStringFromYamlFile;
+import static io.jenkins.plugins.casc.misc.Util.toYamlString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class ConfigAsCodeCertificateTest {
+    private static final String TEST_UPN = "abc@jenkins.com";
+
+    @ClassRule
+    @ConfiguredWithCode("configuration-as-code-certificate-auth.yml")
+    public static JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Rule
+    public LoggerRule l = new LoggerRule().record(MatrixAuthorizationStrategyConfigurator.class, Level.WARNING).capture(20);
+
+    @Test
+    public void should_support_configuration_as_code() {
+        SecurityRealm securityRealm = j.jenkins.getSecurityRealm();
+        assertTrue("security realm", securityRealm instanceof AzureSecurityRealm);
+        AzureSecurityRealm azureSecurityRealm = (AzureSecurityRealm) securityRealm;
+        assertNotEquals("clientId", azureSecurityRealm.getClientIdSecret());
+        assertNotEquals("clientCertificate", azureSecurityRealm.getClientCertificateSecret());
+        assertNotEquals("tenantId", azureSecurityRealm.getTenantSecret());
+        assertEquals("clientId", azureSecurityRealm.getClientId());
+        assertEquals("clientCertificate", azureSecurityRealm.getClientCertificate().getPlainText());
+        assertEquals("Certificate", azureSecurityRealm.getCredentialType());
+        assertEquals("tenantId", azureSecurityRealm.getTenant());
+        assertEquals(0, azureSecurityRealm.getCacheDuration());
+        assertTrue(azureSecurityRealm.isFromRequest());
+
+        AuthorizationStrategy authorizationStrategy = j.jenkins.getAuthorizationStrategy();
```

Comment:
I don't think there's any value from here on in this test as this is testing the authz not authn which is unchanged.

Reply:
I did some more improvements, please review